pasture-algorithms/src/segmentation.rs

```diff
@@ -0,0 +1,313 @@
+use std::vec;
+
+use pasture_core::{
+    containers::{PointBuffer, PointBufferExt},
+    layout::attributes::POSITION_3D,
+    nalgebra::Vector3,
+};
+use rand::Rng;
+use rayon::prelude::*;
+
+/// Represents a line between two points
+/// the ranking shows how many points of the pointcloud are inliers for this specific line
+#[derive(Debug)]
+pub struct Line {
+    first: Vector3<f64>,
+    second: Vector3<f64>,
+    ranking: usize,
+}
+
+/// Represents a plane in coordinate-form: ax + by + cz + d = 0
+/// the ranking shows how many points of the pointcloud are inliers for this specific plane
+#[derive(Debug)]
+pub struct Plane {
+    a: f64,
+    b: f64,
+    c: f64,
+    d: f64,
+    ranking: usize,
+}
+
+/// Ransac Plane Segmentation.
+///
+/// Returns the plane with the highest rating/most inliers and the indices of the inliers.
+///
+/// * `buffer` - The pointcloud-buffer.
+/// * `distance_threshold` - The maximum distance that a point is counted as an inlier.
+/// * `num_of_iterations` - The number of iterations the algorithm performs.
+/// * `parallel` - if true: runs in parallel (using rayon)
+pub fn ransac_plane<T: PointBuffer + Sync>(
```

Comment:
It might be better to publicly expose the `_par` and `_serial` variants of these algorithms. As it stands now, the requirement for `T` to implement `Sync` is unnecessary in the serial case